enterprise/frontend/src/metabase-enterprise/transforms/components/CreateTransformFolderModal/CreateTransformFolderModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { useCallback, useMemo } from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import { useCreateCollectionMutation } from "metabase/api";
+import { getErrorMessage } from "metabase/api/utils";
+import { useToast } from "metabase/common/hooks";
+import {
+  Form,
+  FormErrorMessage,
+  FormProvider,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import * as Errors from "metabase/lib/errors";
+import { Button, Group, Modal, Stack } from "metabase/ui";
+import type { Collection } from "metabase-types/api";
+
+import { TransformCollectionPicker } from "../TransformCollectionPicker";
+
+const COLLECTION_SCHEMA = Yup.object({
+  name: Yup.string()
+    .required(Errors.required)
+    .max(100, Errors.maxLength)
+    .default(""),
+  parent_id: Yup.number().nullable().default(null),
+});
+
+type CollectionFormValues = Yup.InferType<typeof COLLECTION_SCHEMA>;
+
+type CreateTransformFolderModalProps = {
+  onClose: () => void;
+  onCreate?: (collection: Collection) => void;
+};
+
+export function CreateTransformFolderModal({
+  onClose,
+  onCreate,
+}: CreateTransformFolderModalProps) {
+  const [sendToast] = useToast();
+  const [createCollection] = useCreateCollectionMutation();
+
+  const initialValues = useMemo<CollectionFormValues>(
+    () => COLLECTION_SCHEMA.getDefault(),
+    [],
+  );
+
+  const handleSubmit = useCallback(
+    async (values: CollectionFormValues) => {
+      try {
+        const collection = await createCollection({
+          name: values.name,
+          parent_id: values.parent_id,
+          namespace: "transforms",
+        }).unwrap();
+        onCreate?.(collection);
+        onClose();
+      } catch (error) {
+        sendToast({
+          message: getErrorMessage(error, t`Failed to create collection`),
+          icon: "warning",
+        });
+      }
+    },
+    [createCollection, onCreate, onClose, sendToast],
+  );
+
+  return (
+    <Modal title={t`New collection`} opened onClose={onClose} padding="xl">
+      <FormProvider
+        initialValues={initialValues}
+        validationSchema={COLLECTION_SCHEMA}
+        onSubmit={handleSubmit}
+      >
+        <Form>
+          <Stack gap="lg" mt="sm">
+            <FormTextInput
+              name="name"
+              label={t`Name`}
+              placeholder={t`My collection`}
+              data-autofocus
+            />
+            <TransformCollectionPicker
```

Comment:
We currently have a `FormCollectionPicker` input that has the button and collection picker modal. Are we able to use that here instead? I'm not sure what passing the options into it would look like